### ISSUE TITLE
The isFile and isDirectory match types don't seem to work.

### ISSUE THREAD

**Issue by mvarblow** [Bikeman868/UrlRewrite.Net#1]
I'm trying to set up rules for running the angular 2 router in html5 mode. But the rules never fire. It seems that isFile and isDirectory just don't work?

I'm trying to use the following rule to rewrite all requests that are not for static files:

```
<rewrite>
      <rules>
        <rule name="AngularJS Routes" stopProcessing="true">
          <match url=".*" />
          <conditions logicalGrouping="MatchAll">
            <add input="{REQUEST_FILENAME}" matchType="IsFile" negate="true" />
            <add input="{REQUEST_FILENAME}" matchType="IsDirectory" negate="true" />
          </conditions>
          <action type="Rewrite" url="/" />
        </rule>
      </rules>
</rewrite>
```

http://stackoverflow.com/questions/12614072/how-do-i-configure-iis-for-url-rewriting-an-angularjs-application-in-html5-mode

**Comment by Bikeman868** [comment]
Hi Matt,

I see that there is an accepted answer on stackoverflow so I will assume
that you are sorted now.

If you are still not working please feel free to contact me again. It has
only been a couple of months since I wrote UrlRewrite.Net so the code is
fresh in my head and I am sure I can make anything work that you need.

Cheers,
   Martin.

**Comment by Bikeman868** [comment]
I think one of the problems you are having is the isFile and isDirectory are opposite of each other. The isDirectory condition tests if the path has a trailing / and isFile tests that it does not have trailing /. Adding both conditions to one rule will guarantee no match for every request.

If you can describe what you are trying to achieve I am certain it can be achieved with UrlRewrite.Net and if necessary it is easy for me to add new conditions or show you how you can extend it with custom application provided conditions. Either way there is always a solution :-)

**Comment by Bikeman868** [comment]
I did some research and found that the Microsoft rewriter checks for the presence of a physical file when you use the isFile condition. I really want UrlRewrite.Net to be as close to 100% backward compatible as possible so I am changing it to the same behavior.

**Comment by mvarblow** [comment]
Right. The idea in this case is to rewrite the route for any request that does NOT map to either a physical file or folder. 

By the way, thanks for taking the time to write and share this module. It's great to have a version of the rewrite module that I can simply deploy with my application without having to install anything on my servers.

**Comment by Bikeman868** [comment]
I released a new version 1.2.0 that does check for existence of physical file or directory.
Please give it a try and let me know if it's working for you.

By the way, if you attach a debugger (like Visual Studio or DebugView) to your w3p process at startup my rewrite module will output a description of how it interpreted your rules file (check the 'Output' window in Visual Studio). You might find this useful for diagnosing issues. For example it would have told you that the IsFile and IsDirectory were just examining the URL to see if it looked like a file name or a directory name. With version 1.2.0 it will tell you that it is looking for a physical file/directory on disk.

Note also that my url rewriter allows you to pass a lambda expression during initialization and any request for which the lambda is True will output a detailed execution trace showing exactly which rules and conditions were evaluated and what action was taken. I find this pretty useful too.

Martin.
